add optional Kokkos support to magistrate

`distBVH` uses spack packages in CI and it fails when [attempting to serialize `Kokkos::View`](https://github.com/sandialabs/distBVH/actions/runs/10187718320/job/28182385646?pr=26):
```
/opt/view/include/checkpoint/dispatch/dispatch_serializer_nonbyte.h:233:16: error: no matching function for call to 'checkpoint::dispatch::SerializerDispatchNonByte<checkpoint::Sizer, Kokkos::View<unsigned char*, Kokkos::LayoutRight, Kokkos::OpenMP>, checkpoint::dispatch::BasicDispatcher<checkpoint::Sizer, Kokkos::View<unsigned char*, Kokkos::LayoutRight, Kokkos::OpenMP> > >::applyStatic(checkpoint::Sizer&, Kokkos::View<unsigned char*, Kokkos::LayoutRight, Kokkos::OpenMP>*&, checkpoint::SerialSizeType&)'
```

**TODO**:
Add `Kokkos` as an optional dependency and corresponding variant (`default=False`) to `magistrate` package.

`magistrate` enables Kokkos support based on existence of `Kokkos_DIR` or `Kokkos_ROOT`, so I'm assuming something like `"-DKokkos_ROOT={}".format(self.spec["kokkos"].prefix)` will be necessary.

--- a/.github/workflows/test_spack.yml
+++ b/.github/workflows/test_spack.yml
@@ -28,16 +28,26 @@ jobs:
     - name: Checkout code
       uses: actions/checkout@v4
 
+    - name: Prepare Spack configuration
+      run: |
+        mkdir -p /github/home/.spack/
+        cp ci/packages.yaml /github/home/.spack/
+
+        chmod +x ci/test_spack.sh
+
+        mkdir ./kokkos_enabled
+        mkdir ./kokkos_disabled
+
     - name: Run test script (Kokkos enabled)
+      working-directory: ./kokkos_enabled
       env:
         VT_KOKKOS_ENABLED: 1
       run: |
-        chmod +x ci/test_spack.sh
-        ci/test_spack.sh ${{ github.head_ref }}
+        ../ci/test_spack.sh ${{ github.head_ref }}
 
     - name: Run test script (Kokkos disabled)
+      working-directory: ./kokkos_disabled
       env:
         VT_KOKKOS_ENABLED: 0
       run: |
-        chmod +x ci/test_spack.sh
-        ci/test_spack.sh ${{ github.head_ref }}
+        ../ci/test_spack.sh ${{ github.head_ref }}

--- a/.github/workflows/test_spack.yml
+++ b/.github/workflows/test_spack.yml
@@ -28,7 +28,16 @@ jobs:
     - name: Checkout code
       uses: actions/checkout@v4
 
-    - name: Run test script
+    - name: Run test script (Kokkos enabled)
+      env:
+        VT_KOKKOS_ENABLED: 1
+      run: |
+        chmod +x ci/test_spack.sh
+        ci/test_spack.sh ${{ github.head_ref }}
+
+    - name: Run test script (Kokkos disabled)
+      env:
+        VT_KOKKOS_ENABLED: 0
       run: |
         chmod +x ci/test_spack.sh
         ci/test_spack.sh ${{ github.head_ref }}

--- a/ci/packages.yaml
+++ b/ci/packages.yaml
@@ -1,0 +1,5 @@
+packages:
+  openmpi:
+    externals:
+    - spec: openmpi@4.0.4
+      prefix: /usr/local

--- a/ci/test_spack.sh
+++ b/ci/test_spack.sh
@@ -9,7 +9,6 @@ git clone https://github.com/spack/spack.git
 git clone -b "$branch_name" https://github.com/DARMA-tasking/spack-package.git
 
 cd "$spack_path" || exit 1
-git checkout v0.16.3
 cd "$cur_path" || exit 1
 
 declare -A variables_map

--- a/ci/test_spack.sh
+++ b/ci/test_spack.sh
@@ -27,6 +27,7 @@ variables_map["diagnostics_enabled"]="${VT_DIAGNOSTICS_ENABLED:-0}"
 variables_map["diagnostics_runtime_enabled"]="${VT_DIAGNOSTICS_RUNTIME_ENABLED:-0}"
 variables_map["unity_build_enabled"]="${VT_UNITY_BUILD_ENABLED:-0}"
 variables_map["fcontext_enabled"]="${VT_FCONTEXT_ENABLED:-0}"
+variables_map["kokkos_enabled"]="${VT_KOKKOS_ENABLED:-0}"
 
 cmd_vars=()
 for flag in "${!variables_map[@]}"

--- a/ci/test_spack.sh
+++ b/ci/test_spack.sh
@@ -26,7 +26,7 @@ variables_map["diagnostics_enabled"]="${VT_DIAGNOSTICS_ENABLED:-0}"
 variables_map["diagnostics_runtime_enabled"]="${VT_DIAGNOSTICS_RUNTIME_ENABLED:-0}"
 variables_map["unity_build_enabled"]="${VT_UNITY_BUILD_ENABLED:-0}"
 variables_map["fcontext_enabled"]="${VT_FCONTEXT_ENABLED:-0}"
-variables_map["kokkos_enabled"]="${VT_KOKKOS_ENABLED:-0}"
+variables_map["kokkos"]="${VT_KOKKOS_ENABLED:-0}"
 
 cmd_vars=()
 for flag in "${!variables_map[@]}"

--- a/ci/test_spack.sh
+++ b/ci/test_spack.sh
@@ -43,6 +43,7 @@ done
 install_cmd=$(printf " %s" "${cmd_vars[@]}")
 install_cmd="$spack_path/bin/spack install darma-vt@develop build_type=Release ${install_cmd:1}"
 
+"$spack_path"/bin/spack clean --all
 "$spack_path"/bin/spack repo add "$vt_spack_package"
 "$spack_path"/bin/spack external find
 $install_cmd

--- a/packages/darma-magistrate/package.py
+++ b/packages/darma-magistrate/package.py
@@ -43,8 +43,18 @@ class DarmaMagistrate(CMakePackage):
     git = "https://github.com/DARMA-tasking/magistrate.git"
 
     version("develop", branch="develop")
+    variant('kokkos_enabled', default=False, description='Enable Kokkos support')
 
     sanity_check_is_dir = ["include/checkpoint"]
     sanity_check_is_file = ["cmake/magistrateConfig.cmake"]
     sanity_check_is_file = ["cmake/magistrateTargets.cmake"]
     sanity_check_is_file = ["lib/libmagistrate.a"]
+
+    depends_on('kokkos', when='+kokkos_enabled')
+
+    def cmake_args(self):
+        args = [
+            "-DKokkos_ROOT={}".format(self.spec['kokkos'].prefix)
+        ]
+
+        return args

--- a/packages/darma-magistrate/package.py
+++ b/packages/darma-magistrate/package.py
@@ -43,7 +43,7 @@ class DarmaMagistrate(CMakePackage):
     git = "https://github.com/DARMA-tasking/magistrate.git"
 
     version("develop", branch="develop")
-    variant("kokkos_enabled", default=False, description="Enable Kokkos support")
+    variant("kokkos", default=False, description="Enable Kokkos support")
 
     sanity_check_is_dir = ["include/checkpoint"]
     sanity_check_is_file = [
@@ -52,12 +52,12 @@ class DarmaMagistrate(CMakePackage):
         "lib/libmagistrate.a"
     ]
 
-    depends_on("kokkos", when="+kokkos_enabled")
+    depends_on("kokkos", when="+kokkos")
 
     def cmake_args(self):
         args = []
 
-        if "+kokkos_enabled" in self.spec:
+        if "+kokkos" in self.spec:
             args.append("-DKokkos_ROOT={}".format(self.spec['kokkos'].prefix))
 
         return args

--- a/packages/darma-magistrate/package.py
+++ b/packages/darma-magistrate/package.py
@@ -52,7 +52,7 @@ class DarmaMagistrate(CMakePackage):
         "lib/libmagistrate.a"
     ]
 
-    depends_on("kokkos@master", when="+kokkos_enabled")
+    depends_on("kokkos", when="+kokkos_enabled")
 
     def cmake_args(self):
         args = []

--- a/packages/darma-magistrate/package.py
+++ b/packages/darma-magistrate/package.py
@@ -52,7 +52,7 @@ class DarmaMagistrate(CMakePackage):
         "lib/libmagistrate.a"
     ]
 
-    depends_on("kokkos", when="+kokkos_enabled")
+    depends_on("kokkos@master", when="+kokkos_enabled")
 
     def cmake_args(self):
         args = []

--- a/packages/darma-magistrate/package.py
+++ b/packages/darma-magistrate/package.py
@@ -43,18 +43,21 @@ class DarmaMagistrate(CMakePackage):
     git = "https://github.com/DARMA-tasking/magistrate.git"
 
     version("develop", branch="develop")
-    variant('kokkos_enabled', default=False, description='Enable Kokkos support')
+    variant("kokkos_enabled", default=False, description="Enable Kokkos support")
 
     sanity_check_is_dir = ["include/checkpoint"]
-    sanity_check_is_file = ["cmake/magistrateConfig.cmake"]
-    sanity_check_is_file = ["cmake/magistrateTargets.cmake"]
-    sanity_check_is_file = ["lib/libmagistrate.a"]
+    sanity_check_is_file = [
+        "cmake/magistrateConfig.cmake",
+        "cmake/magistrateTargets.cmake",
+        "lib/libmagistrate.a"
+    ]
 
-    depends_on('kokkos', when='+kokkos_enabled')
+    depends_on("kokkos", when="+kokkos_enabled")
 
     def cmake_args(self):
-        args = [
-            "-DKokkos_ROOT={}".format(self.spec['kokkos'].prefix)
-        ]
+        args = []
+
+        if "+kokkos_enabled" in self.spec:
+            args.append("-DKokkos_ROOT={}".format(self.spec['kokkos'].prefix))
 
         return args

--- a/packages/darma-vt/package.py
+++ b/packages/darma-vt/package.py
@@ -151,9 +151,11 @@ class DarmaVt(CMakePackage):
     )
     variant("build_tests", default=False, description="Build all VT tests")
     variant("build_examples", default=False, description="Build all VT examples")
+    variant("kokkos_enabled", default=False, description="Enable Kokkos support")
 
     depends_on("mpi")
-    depends_on("darma-magistrate")
+    depends_on("darma-magistrate+kokkos_enabled", when="+kokkos_enabled")
+    depends_on("darma-magistrate~kokkos_enabled", when="~kokkos_enabled")
 
     sanity_check_is_dir = ["include/vt"]
     sanity_check_is_file = ["cmake/vtConfig.cmake", "cmake/vtTargets.cmake"]

--- a/packages/darma-vt/package.py
+++ b/packages/darma-vt/package.py
@@ -151,11 +151,11 @@ class DarmaVt(CMakePackage):
     )
     variant("build_tests", default=False, description="Build all VT tests")
     variant("build_examples", default=False, description="Build all VT examples")
-    variant("kokkos_enabled", default=False, description="Enable Kokkos support")
+    variant("kokkos", default=False, description="Enable Kokkos support")
 
     depends_on("mpi")
-    depends_on("darma-magistrate+kokkos_enabled", when="+kokkos_enabled")
-    depends_on("darma-magistrate~kokkos_enabled", when="~kokkos_enabled")
+    depends_on("darma-magistrate+kokkos", when="+kokkos")
+    depends_on("darma-magistrate~kokkos", when="~kokkos")
 
     sanity_check_is_dir = ["include/vt"]
     sanity_check_is_file = ["cmake/vtConfig.cmake", "cmake/vtTargets.cmake"]

